docs/Parameters.md

```diff
@@ -54,8 +54,8 @@ The parameter format is `key1=value1 key2=value2 ... ` . And parameters can be s
   * support multi validation data, separate by `,`
 * `num_iterations`, default=`100`, type=int, alias=`num_iteration`,`num_tree`,`num_trees`,`num_round`,`num_rounds`
   * number of boosting iterations
-  * note: `num_tree` here equal with `num_iterations`. For multi-class, it actually learns `num_class * num_iterations` trees.
-  * note: For python/R package, cannot use this parameters to control number of iterations.
+  * note: For python/R package, **this parameters is ignored**.
+  * note: internally, LightGBM constructs `num_class * num_iterations` trees for `multiclass` problems. Neither of aliases of this parameter directly control number of actually constructed trees.
```

Comment:
1. Boosting requires a number of boosting iterations. If there is no iteration, there is no boosting (it becomes a single decision tree in the end). It requires a loop to train a model, therefore no control is provided if not using the mandatory parameter which replaces it. CLI interface is kinda esoteric, therefore the loop is done automatically for us. With the API, this is not true (boosting per iteration).

2. `num_class` still defines the *number of trees per iteration*.

Number of trees != number of boosting iterations. For instance, Random Forest from GBT implementations has a single boosting iteration, but it can have an arbitrary number of trees.

Number of boosting iterations (rounds) = number of steps in gradient descent to perform

Number of trees = total number of trained trees from the beginning to the current round (boosting iteration)